src/promptflow/tests/sdk_cli_test/e2etests/test_flow_run.py

```diff
@@ -150,25 +152,22 @@ def test_run_bulk_error(self, pf):
         assert "Invalid variant format: v, variant should be in format of ${TUNING_NODE.VARIANT}" in str(e.value)
 
     def test_basic_evaluation(self, azure_open_ai_connection: AzureOpenAIConnection, local_client, pf):
-        data_path = f"{DATAS_DIR}/webClassification3.jsonl"
-
         result = pf.run(
-            flow=f"{FLOWS_DIR}/web_classification",
-            data=data_path,
-            column_mapping={"url": "${data.url}"},
+            flow=f"{FLOWS_DIR}/print_env_var",
+            data=f"{DATAS_DIR}/env_var_names.jsonl",
         )
         assert local_client.runs.get(result.name).status == "Completed"
 
         eval_result = pf.run(
             flow=f"{FLOWS_DIR}/classification_accuracy_evaluation",
-            data=data_path,
             run=result.name,
             column_mapping={
-                "groundtruth": "${data.answer}",
-                "prediction": "${run.outputs.category}",
+                "prediction": "${run.outputs.output}",
                 # evaluation reference run.inputs
                 # NOTE: we need this value to guard behavior when a run reference another run's inputs
-                "variant_id": "${run.inputs.url}",
+                "variant_id": "${run.inputs.key}",
+                # can reference other columns in data which doesn't exist in base run's inputs
+                "groundtruth": "${run.inputs.extra_key}",
```

Comment:
why do we change this behavior in the first place? And it looks weird as the usage in the mapping is still `run.inputs.xx` while it's actually not run inputs but run data.

Reply:
I think we can still keep it. As both inputs and outputs are data actually.

Reply:
Agree, if we want to change it, UI also need to change